src/similarity.rs

```diff
@@ -78,9 +76,7 @@ pub trait Analogy {
     /// were present.
     fn analogy_masked(
         &self,
-        word1: &str,
-        word2: &str,
-        word3: &str,
+        query: [&str; 3],
         limit: usize,
         remove: [bool; 3],
```

Comment:
I think it would be best to make `limit` the last argument, so that all word-related arguments are together.